lib/fluent/plugin/in_kafka.rb

```diff
@@ -231,7 +236,18 @@ def consume
 
       messages.each { |msg|
         begin
-          es.add(Fluent::Engine.now, @parser.call(msg, @topic_entry))
+          record = @parser.call(msg, @topic_entry)
+          if @use_record_time
+            if @time_format
+              time_parser = Fluent::TextParser::TimeParser.new(@time_format)
```

Comment:
Creating TimeParser in `each` is very high cost.
It should be created in `configure` when `use_record_time` is `true`